app/src/main/java/com/project/myproject/ui/fragments/ContactsFragment.kt

```diff
@@ -22,35 +33,134 @@ import com.project.myproject.data.models.Contact
         setListeners()
         setObservers()
         setupAnimation()
+    }
 
-        super.onViewCreated(view, savedInstanceState)
+    private fun createNotificationChannel(context: Context) {
+        val name = NOTIFICATION_CHANNEL_NAME
+        val descriptionText = NOTIFICATION_CHANNEL_DESC
+        val importance = NotificationManager.IMPORTANCE_DEFAULT
+        val channel = NotificationChannel(NOTIFICATION_CHANNEL_ID, name, importance).apply {
+            description = descriptionText
+        }
+
+        val notificationManager: NotificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    private fun setupRecyclerView() {
+        binding.rvContacts.apply {
+            setupAdapterScroll(this)
+            adapter = this@ContactsFragment.adapter
+            layoutManager = LinearLayoutManager(requireContext())
+            addItemDecoration(DefaultItemDecorator(resources.getDimensionPixelSize(R.dimen.contacts_item_margin)))
+
+            val itemTouchHelper = ItemTouchHelper(SwipeToDeleteCallback { contact ->
+                viewModel.deleteContact(contact.id)
+                showDeleteSnackbar(contact)
+            })
+
+            itemTouchHelper.attachToRecyclerView(this)
+        }
+        viewModel.fetchContacts()
+        collectContactsData()
+    }
+
+    private fun collectContactsData() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.contacts.collect { contacts ->
+                adapter.submitList(contacts)
+            }
+        }
+    }
+
+    private fun setupSearchFunctionality() {
+        setupSearchButtonListeners()
+    }
+
+    private fun setupSearchButtonListeners() {
+        binding.ivToolbarSearch.setOnClickListener {
+            if (ActivityCompat.checkSelfPermission(requireContext(), Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+                requestNotificationPermission()
+            } else {
+                showSearchNotification(requireContext())
+            }
+        }
+    }
+
+    private fun requestNotificationPermission() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            requestPermissions(arrayOf(Manifest.permission.POST_NOTIFICATIONS),
+                Constants.NOTIFICATION_PERMISSION_REQUEST_CODE
+            )
+        }
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == Constants.NOTIFICATION_PERMISSION_REQUEST_CODE) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                showSearchNotification(requireContext())
+            }
+        }
+    }
```

Comment:
Bad approach. Try [this](https://developer.android.com/training/permissions/requesting).